File: soft uni prgramming fundamentals/6. Lists/Lists/src/BombNumbers.java
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class BombNumbers {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        List<Integer> list = Arrays.stream(s.nextLine().split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());

        int[] specialNumber = Arrays.stream(s.nextLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int bombNumber = specialNumber[0];
        int power = specialNumber[1];
        int sum = 0;

            for (int i = 0; i < list.size(); i++) {
                if (list.get(i).equals(bombNumber)) {

                        for (int j = Math.max(i - power, 0); j <= Math.min(i + power, list.size() - 1); j++){
                            int temp=0;
                            list.set(j,0);
                        }

                }
            }



            for (int num : list) {
                sum += num;
            }
            System.out.println(sum);
        }
    }

File: soft uni prgramming fundamentals/6. Lists/Lists C#/Lists/BombNumbers/BombNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace BombNumbers
{
    class BombNumbers
    {
        static void Main(string[] args)
        {
            List<int> numbers = Console.ReadLine().Split().Select(int.Parse).ToList();
            List<int> magicNumbers = Console.ReadLine().Split().Select(int.Parse).ToList();
            int bomb = magicNumbers[0];
            int power = magicNumbers[1];
            int sum = 0;


            for (int i = 0; i < numbers.Count(); i++)
            {
                if (numbers[i]==bomb)
                {

                    for (int j = Math.Max(i - power, 0); j <= Math.Min(i + power, numbers.Count() - 1); j++)
                    {
                        int temp = 0;
                        numbers[j]= 0;
                    }

                }
            }



            foreach (var num in numbers)           
            {
                sum += num;
            }
            Console.WriteLine(sum);
        }
        }
    }

